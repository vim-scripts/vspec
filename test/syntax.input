" Common set up  "{{{1

silent filetype plugin on
syntax enable

function! SynStack(lnum, col)
  return map(synstack(a:lnum, a:col), 'synIDattr(v:val, "name")')
endfunction




function! s:describe__highlighting()  "{{{1
  tabnew
  tabonly!

  silent put =[
  \   'It should be highlighted.',
  \   'Should 123 be true',
  \   'Should string('''') ==# ''''',
  \ ]
  1 delete _

  setfiletype vim

  It should highlight :It properly.

  Should SynStack(1, 1) ==# ['vimVspecCommand']
  Should SynStack(1, 2) ==# ['vimVspecCommand']
  Should SynStack(1, 3) ==# []
  Should SynStack(1, 4) ==# ['vimVspecItArgument']
  Should SynStack(1, 5) ==# ['vimVspecItArgument']
  Should SynStack(1, 6) ==# ['vimVspecItArgument']
  Should SynStack(1, 7) ==# ['vimVspecItArgument']
  Should SynStack(1, 8) ==# ['vimVspecItArgument']
  Should SynStack(1, 9) ==# ['vimVspecItArgument']
  Should SynStack(1, 10) ==# ['vimVspecItArgument']
  Should SynStack(1, 11) ==# ['vimVspecItArgument']
  Should SynStack(1, 12) ==# ['vimVspecItArgument']
  Should SynStack(1, 13) ==# ['vimVspecItArgument']
  Should SynStack(1, 14) ==# ['vimVspecItArgument']
  Should SynStack(1, 15) ==# ['vimVspecItArgument']
  Should SynStack(1, 16) ==# ['vimVspecItArgument']
  Should SynStack(1, 17) ==# ['vimVspecItArgument']
  Should SynStack(1, 18) ==# ['vimVspecItArgument']
  Should SynStack(1, 19) ==# ['vimVspecItArgument']
  Should SynStack(1, 20) ==# ['vimVspecItArgument']
  Should SynStack(1, 21) ==# ['vimVspecItArgument']
  Should SynStack(1, 22) ==# ['vimVspecItArgument']
  Should SynStack(1, 23) ==# ['vimVspecItArgument']
  Should SynStack(1, 24) ==# ['vimVspecItArgument']
  Should SynStack(1, 25) ==# ['vimVspecItArgument']

  It should highlight :Should properly.

  Should SynStack(2, 1) ==# ['vimVspecCommand']
  Should SynStack(2, 2) ==# ['vimVspecCommand']
  Should SynStack(2, 3) ==# ['vimVspecCommand']
  Should SynStack(2, 4) ==# ['vimVspecCommand']
  Should SynStack(2, 5) ==# ['vimVspecCommand']
  Should SynStack(2, 6) ==# ['vimVspecCommand']
  Should SynStack(2, 7) ==# []
  Should SynStack(2, 8) ==# ['vimNumber']
  Should SynStack(2, 9) ==# ['vimNumber']
  Should SynStack(2, 10) ==# ['vimNumber']
  Should SynStack(2, 11) ==# []
  " Should SynStack(2, 12) ==# []
  " Should SynStack(2, 13) ==# []
  " Should SynStack(2, 14) ==# []
  " Should SynStack(2, 15) ==# []
  " Should SynStack(2, 16) ==# []
  " Should SynStack(2, 17) ==# []
  " Should SynStack(2, 18) ==# []

  Should SynStack(3, 1) ==# ['vimVspecCommand']
  Should SynStack(3, 2) ==# ['vimVspecCommand']
  Should SynStack(3, 3) ==# ['vimVspecCommand']
  Should SynStack(3, 4) ==# ['vimVspecCommand']
  Should SynStack(3, 5) ==# ['vimVspecCommand']
  Should SynStack(3, 6) ==# ['vimVspecCommand']
  Should SynStack(3, 7) ==# []
  Should SynStack(3, 8) ==# ['vimFunc', 'vimFuncName']
  Should SynStack(3, 9) ==# ['vimFunc', 'vimFuncName']
  Should SynStack(3, 10) ==# ['vimFunc', 'vimFuncName']
  Should SynStack(3, 11) ==# ['vimFunc', 'vimFuncName']
  Should SynStack(3, 12) ==# ['vimFunc', 'vimFuncName']
  Should SynStack(3, 13) ==# ['vimFunc', 'vimFuncName']
  Should SynStack(3, 14) ==# ['vimOperParen', 'vimOper']
  Should SynStack(3, 15) ==# ['vimOperParen', 'vimString']
  Should SynStack(3, 16) ==# ['vimOperParen', 'vimString']
  Should SynStack(3, 17) ==# ['vimOper']
  Should SynStack(3, 18) ==# []
  Should SynStack(3, 19) ==# ['vimOper']
  Should SynStack(3, 20) ==# ['vimOper']
  Should SynStack(3, 21) ==# ['vimOper']
  Should SynStack(3, 22) ==# []
  Should SynStack(3, 23) ==# ['vimString']
  Should SynStack(3, 24) ==# ['vimString']
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
