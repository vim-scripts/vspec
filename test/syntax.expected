==== highlighting 
---- It should highlight :It properly. 
.........................
---- It should highlight :Should properly. 
...................................



**** Result ****


60 examples, 0 failures
